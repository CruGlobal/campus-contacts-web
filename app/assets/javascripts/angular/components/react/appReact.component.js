import template from './appReact.html';

angular.module('missionhubApp').component('appReact', {
    controller: AppReactController,
    template: template,
});

function AppReactController($rootScope) {}
