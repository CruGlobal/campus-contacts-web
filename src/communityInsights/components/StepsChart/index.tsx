import { useQuery } from 'react-apollo-hooks';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { DocumentNode } from 'graphql';

import BarChart from '../BarChart';

const Wrapper = styled.div`
    height: 400px;
    position: relative;
`;

interface Props {
    query: DocumentNode;
    mapData: (data: any) => any;
    variables: any;
    label: string;
    index: string;
    nullContent: string;
}
const StepsChart = ({
    query,
    mapData,
    label,
    index,
    variables,
    nullContent,
}: Props) => {
    const { data, loading } = useQuery(query, { variables });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Wrapper>{t('loading')}</Wrapper>;
    }

    return (
        <Wrapper>
            <BarChart
                nullContent={nullContent}
                data={mapData(data)}
                keys={[label]}
                indexBy={index}
                onFilterChanged={() => {}}
                index={0}
                filterType={'month'}
            />
        </Wrapper>
    );
};

export default StepsChart;
