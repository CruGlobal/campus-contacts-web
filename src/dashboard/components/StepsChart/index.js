import { useQuery } from 'react-apollo-hooks';
import React from 'react';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';

import BarChart from '../BarChart';

const Wrapper = styled.div`
    height: 400px;
    position: relative;
`;

const StepsChart = ({ query, mapData, label, index, variables }) => {
    const { data, loading } = useQuery(query, { variables });
    const { t } = useTranslation('insights');

    if (loading) {
        return <Wrapper>{t('loading')}</Wrapper>;
    }

    return (
        <Wrapper>
            <BarChart data={mapData(data)} keys={[label]} indexBy={index} />
        </Wrapper>
    );
};

export default StepsChart;
