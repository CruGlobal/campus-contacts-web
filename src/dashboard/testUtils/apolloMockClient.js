import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
import { buildClientSchema } from 'graphql/utilities';
import { createMockerFromIntrospection } from 'fraql/mock';
import gql from 'fraql';

import introspectionQuery from '../../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument, options) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
