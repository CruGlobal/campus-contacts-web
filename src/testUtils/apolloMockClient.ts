import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema } from 'graphql-tools';
// @ts-ignore
import { buildClientSchema } from 'graphql/utilities';
// @ts-ignore
import { createMockerFromIntrospection } from 'fraql/mock';
// @ts-ignore
import gql from 'fraql';

import introspectionQuery from '../../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks = {}) => {
    const schema = buildClientSchema(introspectionQuery);

    addMockFunctionsToSchema({
        schema,
        mocks: { ...globalMocks, ...mocks },
    });

    return new ApolloClient({
        cache: new InMemoryCache(),
        link: new SchemaLink({ schema }),
    });
};

export const mockFragment = (fragmentDocument: any, options: any) => {
    return createMockerFromIntrospection(introspectionQuery, {
        mocks: globalMocks,
    }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
