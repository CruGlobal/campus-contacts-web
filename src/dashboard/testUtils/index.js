import React, { ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';

import { createApolloMockClient } from './apolloMockClient';
import defaultTheme from '../defaultTheme';
import { ThemeProvider } from 'emotion-theming';
import AppContext from '../appContext';

export const renderWithContext = (component, { mocks } = {}, context = {}) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={context}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });

    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
