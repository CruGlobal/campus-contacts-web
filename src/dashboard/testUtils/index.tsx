import React, { ReactNode } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';

import defaultTheme from '../defaultTheme';
import AppContext from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

export const renderWithContext = (
    component: any,
    { mocks: mocks = {}, appContext = {} } = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    // @ts-ignore
    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
