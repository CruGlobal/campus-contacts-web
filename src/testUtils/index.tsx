import React, { ReactNode, ReactElement } from 'react';
import { ApolloProvider } from 'react-apollo-hooks';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import { IMocks } from 'graphql-tools';

import defaultTheme from '../defaultTheme';
import {
    AppContext,
    AppContextValue,
    appContextDefaultValue,
} from '../appContext';

import { createApolloMockClient } from './apolloMockClient';

interface RenderWithContextParams {
    mocks?: IMocks;
    appContext?: AppContextValue;
}

export const renderWithContext = (
    component: ReactElement,
    {
        mocks: mocks = {},
        appContext = appContextDefaultValue,
    }: RenderWithContextParams = {},
) => {
    const mockApolloClient = createApolloMockClient(mocks);

    // Warning: don't call any functions in here that return new instances on every call.
    // All the props need to stay the same otherwise renderer won't work.
    const wrapper = ({ children }: { children?: ReactNode }) => (
        <ApolloProvider client={mockApolloClient}>
            <AppContext.Provider value={appContext}>
                <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
            </AppContext.Provider>
        </ApolloProvider>
    );

    const renderResult = render(component, { wrapper });
    return {
        ...renderResult,
        snapshot: () => {
            expect(renderResult.container).toMatchSnapshot();
        },
    };
};
