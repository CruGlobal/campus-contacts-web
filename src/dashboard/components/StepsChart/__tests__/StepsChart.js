import React from 'react';
import gql from 'graphql-tag';
import moment from 'moment';
import { waitForElement } from '@testing-library/react';

import { renderWithContext } from '../../../testUtils';
import StepsChart from '../';

const QUERY = gql`
    query organizationPathwayStagesReport(
        $period: String!
        $organizationId: ID!
        $endDate: ISO8601DateTime!
    ) {
        organizationPathwayStagesReport(
            period: $period
            organizationId: $organizationId
            endDate: $endDate
        ) {
            stage {
                name
            }
            stepsAddedCount
        }
    }
`;

describe('<StepsChart />', () => {
    it('should render properly', async () => {
        renderWithContext(
            <StepsChart
                query={QUERY}
                mapData={data =>
                    data.organizationPathwayStagesReport.map(row => ({
                        ['label']: row.stepsAddedCount,
                        ['index']: row.stage.name.toUpperCase(),
                    }))
                }
                label={'label'}
                index={'index'}
                variables={{
                    period: '',
                    organizationId: 1,
                    endDate: moment('2019-07-21').format(),
                }}
            />,
        ).snapshot();
    });

    it('should render properly with data', async () => {
        const { snapshot, getByText } = renderWithContext(
            <StepsChart
                query={QUERY}
                mapData={data =>
                    data.organizationPathwayStagesReport.map(row => ({
                        ['label']: row.stepsAddedCount,
                        ['index']: row.stage.name.toUpperCase(),
                    }))
                }
                label={'label'}
                index={'index'}
                variables={{
                    period: '',
                    organizationId: 1,
                    endDate: moment('2019-07-21').format(),
                }}
            />,
            {
                mocks: {
                    Query: () => ({
                        organizationPathwayStagesReport: () => [
                            {
                                stepsAddedCount: 20,
                                stage: {
                                    name: 'NO STAGE',
                                },
                            },
                            {
                                stepsAddedCount: 25,
                                stage: {
                                    name: 'UNINTERESTED',
                                },
                            },
                        ],
                    }),
                },
                appContext: {
                    orgId: 1,
                },
            },
        );

        await waitForElement(() => getByText('NO STAGE'));
        snapshot();
    });
});
