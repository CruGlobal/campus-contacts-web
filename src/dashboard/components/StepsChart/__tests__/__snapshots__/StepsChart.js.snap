// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepsChart /> should render properly 1`] = `
<div>
  <div
    class="css-1a3wyyp"
  >
    Loading...
  </div>
</div>
`;

exports[`<StepsChart /> should render properly with data 1`] = `
<div>
  <div
    class="css-1a3wyyp"
  >
    <div>
      <div
        class="css-1a3wyyp"
      >
        <div
          style="position: relative;"
        >
          <svg
            height="400"
            role="img"
            width="848"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="transparent"
              height="400"
              width="848"
            />
            <g
              transform="translate(30,30)"
            >
              <g>
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="280"
                  y2="280"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="258"
                  y2="258"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="235"
                  y2="235"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="213"
                  y2="213"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="190"
                  y2="190"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="168"
                  y2="168"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="146"
                  y2="146"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="123"
                  y2="123"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="101"
                  y2="101"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="78"
                  y2="78"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="56"
                  y2="56"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="34"
                  y2="34"
                />
                <line
                  opacity="1"
                  stroke="#ECEEF2"
                  stroke-width="1"
                  x1="0"
                  x2="818"
                  y1="11"
                  y2="11"
                />
              </g>
              <g
                transform="translate(0,280)"
              >
                <g
                  style="opacity: 1;"
                  transform="translate(239,0)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="text-before-edge"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="middle"
                    transform="translate(0,20) rotate(0)"
                  >
                    NO STAGE
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(579,0)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="text-before-edge"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="middle"
                    transform="translate(0,20) rotate(0)"
                  >
                    UNINTERESTED
                  </text>
                </g>
                <line
                  style="stroke: transparent; stroke-width: 1;"
                  x1="0"
                  x2="818"
                  y1="0"
                  y2="0"
                />
              </g>
              <g
                transform="translate(0,0)"
              >
                <g
                  style="opacity: 1;"
                  transform="translate(0,280)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    0
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,258)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    2
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,235)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    4
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,213)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    6
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,190)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    8
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,168)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    10
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,146)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    12
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,123)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    14
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,101)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    16
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,78)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    18
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,56)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    20
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,34)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    22
                  </text>
                </g>
                <g
                  style="opacity: 1;"
                  transform="translate(0,11)"
                >
                  <line
                    style="stroke: #777777; stroke-width: 1;"
                    x1="0"
                    x2="0"
                    y1="0"
                    y2="0"
                  />
                  <text
                    dominant-baseline="central"
                    style="font-weight: 600; letter-spacing: 1px; font-family: Source Sans Pro; font-size: 10px; fill: #B4B6BA;"
                    text-anchor="end"
                    transform="translate(-5,0) rotate(0)"
                  >
                    24
                  </text>
                </g>
                <line
                  style="stroke: transparent; stroke-width: 1;"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="280"
                />
              </g>
              <g>
                <g
                  transform="translate(137, 56)"
                >
                  <rect
                    fill="#3CC8E6"
                    height="224"
                    rx="0"
                    ry="0"
                    stroke="#3CC8E6"
                    stroke-width="0"
                    width="204"
                  />
                </g>
                <g
                  transform="translate(477, 0)"
                >
                  <rect
                    fill="#3CC8E6"
                    height="280"
                    rx="0"
                    ry="0"
                    stroke="#3CC8E6"
                    stroke-width="0"
                    width="204"
                  />
                </g>
              </g>
            </g>
          </svg>
        </div>
      </div>
      <div
        class="css-1hr8g7h"
      />
    </div>
  </div>
</div>
`;
